src/jsxc.lib.xmpp.chatState.js

```diff
@@ -0,0 +1,247 @@
+
+      return;
+   }
+
+   self.conn = jsxc.xmpp.conn;
+
+   $(document).on('composing.chatstates', jsxc.xmpp.chatState.onComposing);
+   $(document).on('paused.chatstates', jsxc.xmpp.chatState.onPaused);
+   $(document).on('active.chatstates', jsxc.xmpp.chatState.onActive);
+};
+
+/**
+ * Composing event received. Display message.
+ *
+ * @memberOf jsxc.xmpp.chatState
+ * @param  {Event} ev
+ * @param  {String} jid
+ */
+jsxc.xmpp.chatState.onComposing = function(ev, jid) {
+   var self = jsxc.xmpp.chatState;
+   var bid = jsxc.jidToBid(jid);
+   var data = jsxc.storage.getUserItem('buddy', bid) || null;
+
+   if (!data || jsxc.xmpp.chatState.isDisabled()) {
+      return;
+   }
+
+   // ignore own notifications in groupchat
+   if (data.type === 'groupchat' &&
+      Strophe.getResourceFromJid(jid) === Strophe.getNodeFromJid(self.conn.jid)) {
+      return;
+   }
+
+   var user = data.type === 'groupchat' ? Strophe.getResourceFromJid(jid) : data.name;
+   var win = jsxc.gui.window.get(bid);
+
+   if (win.length === 0) {
+      return;
+   }
+
+   clearTimeout(win.data('composing-timeout'));
+
+   // add user in array if necessary
+   var usersComposing = win.data('composing') || [];
+   if (usersComposing.indexOf(user) === -1) {
+      usersComposing.push(user);
+      win.data('composing', usersComposing);
+   }
+
+   var textarea = win.find('.jsxc_textarea');
+   var composingNotif = textarea.find('.jsxc_composing');
+
+   if (composingNotif.length < 1) {
+      // notification not present, add it
+      composingNotif = $('<div>').addClass('jsxc_composing')
+         .addClass('jsxc_chatmessage')
+         .addClass('jsxc_sys')
+         .appendTo(textarea);
+   }
+
+   var msg = self._genComposingMsg(usersComposing);
+   composingNotif.text(msg);
+
+   // scroll to bottom
+   jsxc.gui.window.scrollDown(bid);
+
+   // show message
+   composingNotif.addClass('jsxc_fadein');
+};
+
+/**
+ * Pause event receive. Remove or update composing message.
+ *
+ * @memberOf jsxc.xmpp.chatState
+ * @param  {Event} ev
+ * @param  {String} jid
+ */
+jsxc.xmpp.chatState.onPaused = function(ev, jid) {
+   var self = jsxc.xmpp.chatState;
+   var bid = jsxc.jidToBid(jid);
+   var data = jsxc.storage.getUserItem('buddy', bid) || null;
+
+   if (!data || jsxc.xmpp.chatState.isDisabled()) {
+      return;
+   }
+
+   var user = data.type === 'groupchat' ? Strophe.getResourceFromJid(jid) : data.name;
+   var win = jsxc.gui.window.get(bid);
+
+   if (win.length === 0) {
+      return;
+   }
+
+   var el = win.find('.jsxc_composing');
+   var usersComposing = win.data('composing') || [];
+
+   if (usersComposing.indexOf(user) >= 0) {
+      // remove user from list
+      usersComposing.splice(usersComposing.indexOf(user), 1);
+      win.data('composing', usersComposing);
+   }
+
+   if (usersComposing.length === 0) {
+      var duration = parseFloat(el.css('transition-duration'));
+
+      if (el.css('transition-duration').match(/s$/)) {
```

Comment:
I will have a look into the duration issue. Could you open another issue for the translation problem?